src/Task/Git/AbstractRegex.php

```diff
@@ -0,0 +1,53 @@
+<?php
+
+namespace GrumPHP\Task\Git;
+
+use GrumPHP\Configuration\GrumPHP;
+use GrumPHP\Task\TaskInterface;
+use Symfony\Component\OptionsResolver\OptionsResolver;
+
+abstract class AbstractRegex implements TaskInterface
```

Comment:
Currently I don't think this abstract class is the way to go. There are some people working on the `git_commit_message` task in e.g. #319. This will surely break the implementation and doesn't add much value since it only contains the configurable options. I think it's better to drop this class and create a separate implementation.